### ISSUE TITLE
Incompat with packaging==22.0

### ISSUE THREAD

**Issue by tnh8** [wimglenn/johnnydep#121]
Using:
```
$ johnnydep --version
johnnydep v1.20.0
pip --version
pip 23.1.2 from C:\Python37\lib\site-packages\pip (python 3.7)
$ python --version
Python 3.7.9
```
and

trying:
```
$ johnnydep robotframework-robocop
```

gives:
```
2023-05-25 05:51:43 [info     ] init johnnydist                [johnnydep.lib] dist=robotframework-robocop parent=None
Traceback (most recent call last):
  File "C:\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python37\Scripts\johnnydep.exe\__main__.py", line 7, in <module>
  File "C:\Python37\lib\site-packages\johnnydep\cli.py", line 94, in main
    recurse=args.recurse,
  File "C:\Python37\lib\site-packages\johnnydep\lib.py", line 303, in serialise
    return tabulate.tabulate(table, headers="keys")
  File "C:\Python37\lib\site-packages\tabulate.py", line 1427, in tabulate
    tabular_data, headers, showindex=showindex
  File "C:\Python37\lib\site-packages\tabulate.py", line 1025, in _normalize_tabular_data
    rows = list(tabular_data)
  File "C:\Python37\lib\site-packages\johnnydep\lib.py", line 343, in gen_table
    for prefix, _fill, dist in anytree.RenderTree(johnnydist):
  File "C:\Python37\lib\site-packages\anytree\render.py", line 282, in __next
    children = node.children
  File "C:\Python37\lib\site-packages\johnnydep\lib.py", line 146, in children
    for dep in self.requires:
  File "C:\Python37\lib\site-packages\johnnydep\lib.py", line 125, in requires
    if req.marker.evaluate(dict(env_data, extra=extra)):
  File "C:\Python37\lib\site-packages\packaging\markers.py", line 241, in evaluate
    return _evaluate_markers(self._markers, current_environment)
  File "C:\Python37\lib\site-packages\packaging\markers.py", line 150, in _evaluate_markers
    lhs_value, rhs_value = _normalize(lhs_value, rhs_value, key=environment_key)
  File "C:\Python37\lib\site-packages\packaging\markers.py", line 124, in _normalize
    return tuple(canonicalize_name(v) for v in values)
  File "C:\Python37\lib\site-packages\packaging\markers.py", line 124, in <genexpr>
    return tuple(canonicalize_name(v) for v in values)
  File "C:\Python37\lib\site-packages\packaging\utils.py", line 34, in canonicalize_name
    value = _canonicalize_regex.sub("-", name).lower()
TypeError: expected string or bytes-like object
```

**Comment by wimglenn** [comment]
I can not reproduce this on Linux. I thought it might be some weirdness about older Python on Windows, but I can not reproduce it there either:

```
(issue121) C:\Users\wglenn>python --version
Python 3.7.9

(issue121) C:\Users\wglenn>pip --version
pip 23.1.2 from C:\Users\wglenn\AppData\Local\miniconda3\envs\issue121\lib\site-packages\pip (python 3.7)

(issue121) C:\Users\wglenn>johnnydep --version
johnnydep v1.20.0

(issue121) C:\Users\wglenn>johnnydep robotframework-robocop -v0
name                        summary
--------------------------  ----------------------------------------------------------------------------------------
robotframework-robocop      Static code analysis tool (linter) for Robot Framework
├── jinja2<4.0,>=3.0        A very fast and expressive template engine.
│   └── MarkupSafe>=2.0     Safely add untrusted strings to HTML/XML markup.
├── pathspec<0.12,>=0.9     Utility library for gitignore style pattern matching of file paths.
├── python-dateutil>=2.8.2  Extensions to the standard Python datetime module
│   └── six>=1.5            Python 2 and 3 compatibility utilities
├── pytz>=2022.7            World timezone definitions, modern and historical
├── robotframework>=3.2.2   Generic automation framework for acceptance testing and robotic process automation (RPA)
└── tomli>=2.0.0            A lil' TOML parser
```

Can you make a reproducible example? What's different about your environment?

**Comment by wimglenn** [comment]
Yes, I think it is due to older packaging v22.0 dependency. Could you try with packaging >= 23 ?

**Comment by tnh8** [comment]
Yes, installing `packaging==23.0` solves both examples I have posted in this thread. Thanks!

Some dependency requirements update needed?

I am satisfied, feel free to close the issue.

**Comment by wimglenn** [comment]
Yes, root cause is https://github.com/pypa/packaging/issues/638 fixed in packaging 23.0
